File: oscexchange/src/main/java/ac/at/fhstp/digitech/oscexchange/OSCReceiveRequest.java
package ac.at.fhstp.digitech.oscexchange;

abstract class OSCReceiveRequest extends OSCRequest {

    private final OSCArgsValidator validator;


    protected OSCReceiveRequest(OSCAddress address, OSCArgsValidator validator, OSCRequest next) {
        super(address, next);
        this.validator = validator;
    }

    protected OSCReceiveRequest(OSCAddress address, OSCArgsValidator validator) {
        super(address);
        this.validator = validator;
    }

}

File: oscexchange/src/main/java/ac/at/fhstp/digitech/oscexchange/OSCReceiveRequest.kt
package ac.at.fhstp.digitech.oscexchange

internal abstract class OSCReceiveRequest : OSCRequest {
    val validator: OSCArgsValidator?

    protected constructor(
        address: OSCAddress,
        validator: OSCArgsValidator?
    ) : super(address) {
        this.validator = validator
    }

    protected constructor(address: OSCAddress) : super(address) {
        validator = null
    }
}